ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -342,6 +391,181 @@ objects:
+          SURVIVING_MASTER_NODE=$(oc get ${MAPI_POD} -n openshift-machine-api -o jsonpath='{.spec.nodeName}')
+          mapfile -t MASTER_NODES_TO_REMOVE < <(oc get nodes -l node-role.kubernetes.io/master= -o name | grep -v "${SURVIVING_MASTER_NODE}")
+          MASTER_MACHINES_TO_REMOVE=()
+          for master in ${MASTER_NODES_TO_REMOVE[@]}
+          do
+            MASTER_MACHINES_TO_REMOVE+=($(oc get ${master} -o jsonpath='{.metadata.annotations.machine\.openshift\.io\/machine}' | cut -d '/' -f 2))
+          done
+
+          echo "Destroy two masters"
+          # Scale down etcd quorum guard
+          oc scale --replicas=0 deployment.apps/etcd-quorum-guard -n openshift-machine-config-operator
+
+          for machine in ${MASTER_MACHINES_TO_REMOVE[@]}
+          do
+            retry 10 oc --request-timeout=5s -n openshift-machine-api delete machine ${machine}
+          done
+
+          echo "Confirm meltdown"
+          sleep 30
+          oc --request-timeout=5s get nodes && exit 1
+
+          echo "Restore single etcd - /usr/local/bin/etcd-snapshot-restore.sh"
+          bastion_ssh core@${SURVIVING_MASTER_NODE} "sudo -i /bin/bash -x /usr/local/bin/etcd-snapshot-restore.sh"
+
+          echo "Wait for API server to come up"
+          retry 30 oc get nodes
+
+          # Workaround for https://bugzilla.redhat.com/show_bug.cgi?id=1707006
+          echo "Restart SDN"
+          retry 10 oc delete pods -l app=sdn -n openshift-sdn --wait=false
+
+          echo "Create two masters via Machine API"
+          retry 10 oc get machines -n openshift-machine-api
+          # Clone existing masters, update IDs and oc apply
+          SURVIVING_MASTER_MACHINE=$(oc get machine -l machine.openshift.io/cluster-api-machine-role=master -n openshift-machine-api -o name | cut -d '/' -f 2)
+          SURVIVING_MASTER_NUM=${SURVIVING_MASTER_MACHINE##*-}
+          SURVIVING_MASTER_PREFIX=${SURVIVING_MASTER_MACHINE%-*}
+          retry 10 oc get --export machine ${SURVIVING_MASTER_MACHINE} -n openshift-machine-api -o yaml > /tmp/machine.template
+          SURVIVING_MASTER_AZ=$(grep -oP 'availabilityZone: (\K.+)' /tmp/machine.template)
+          SURVIVING_MASTER_AZ_INDEX=$(grep -oP 'availabilityZone: .*(\K.)' /tmp/machine.template)
+
+          MASTER_INDEX=0
+          for i in $(seq 0 1); do
+            if [[ "${MASTER_INDEX}" == "${SURVIVING_MASTER_NUM}" ]]; then MASTER_INDEX=$((MASTER_INDEX+1)); fi
+            cat /tmp/machine.template \
+              | sed 's;selfLink.*;;g' \
+              | sed "s;name: ${SURVIVING_MASTER_PREFIX}-${SURVIVING_MASTER_NUM};name: ${SURVIVING_MASTER_PREFIX}-${MASTER_INDEX};" > /tmp/machine_${i}.yaml
+              retry 10 oc create -n openshift-machine-api -f /tmp/machine_${i}.yaml
+              MASTER_INDEX=$((MASTER_INDEX+1))
+          done
+
+          echo "Waiting for machines to be created"
+          NEW_MASTER_IPS=()
+          for i in $(seq 0 60); do
```

Comment:
Why not use retry.

Reply:
We need to break out of the loop when all three masters got the IP, so wrapping a large `oc` call and comparison in a single command would be unreadable